tests/testParallelCatalogWriter.py

```diff
@@ -0,0 +1,246 @@
+from __future__ import with_statement
+import unittest
+import sqlite3
+import os
+import numpy as np
+
+import lsst.utils.tests
+from lsst.utils import getPackageDir
+from lsst.sims.catalogs.definitions import parallelCatalogWriter
+from lsst.sims.catalogs.definitions import InstanceCatalog
+from lsst.sims.catalogs.decorators import compound, cached
+from lsst.sims.catalogs.db import CatalogDBObject
+
+
+def setup_module(module):
+    lsst.utils.tests.init()
+
+
+class DbClass(CatalogDBObject):
+    tableid = 'test'
+    database = os.path.join(getPackageDir('sims_catalogs'),
+                            'tests', 'scratchSpace', 'parallel_test_db.db')
+
+    host = None
+    port = None
+    driver = 'sqlite'
+    objid = 'paralle_writer_test_db'
+    idColKey = 'id'
+
+class CatClass1(InstanceCatalog):
+    column_outputs = ['id', 'test1', 'ii']
+    cannot_be_null = ['valid1']
+
+    @compound('test1', 'valid1')
+    def get_values(self):
+        ii = self.column_by_name('ii')
+        return np.array([self.column_by_name('id')**2,
+                         np.where(ii%2==1, ii, None)])
+
+class CatClass2(InstanceCatalog):
+    column_outputs = ['id', 'test2', 'ii']
+    cannot_be_null = ['valid2']
+
+    @compound('test2', 'valid2')
+    def get_values(self):
+        ii = self.column_by_name('id')
+        return np.array([self.column_by_name('id')**3,
+                         np.where(ii%2==1, ii, None)])
+
+class CatClass3(InstanceCatalog):
+    column_outputs = ['id', 'test3', 'ii']
+    cannot_be_null = ['valid3']
+
+    @cached
+    def get_test3(self):
+        return self.column_by_name('id')**4
+
+    @cached
+    def get_valid3(self):
+        ii = self.column_by_name('id')
+        return np.where(ii%5 == 0, ii, None)
+
```

Comment:
This should probably be work for a future ticket.  It is not trivial since `CompoundInstanceCatalog` allows you to pass in `InstanceCatalog`s that use potentially different database connections and then does the work of grouping them by like connection.  `parallelCatalogWriter` assumes that everything you pass it relies on the same database connection.